File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
import math
import sys
input=sys.stdin.readline

def solution(board, h, w):
    answer = 0
    n=len(board)

    dh=[0,1,-1,0]    #h축 움직임
    dw=[1,0,0,-1]    #w축 움직임

    for i in range(4):
        h_check=h+dh[i]
        w_check=w+dw[i]
        if(0<=h_check<n and 0<=w_check<n):
            if(board[h][w]== board[h_check][w_check]):
                answer+=1


    return answer

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
class Solution {
    public int solution(String[][] board, int h, int w) {

        String color=board[h][w];
        int cnt=0;
        // 상하좌우
        int[] dh={-1,1,0,0};
        int[] dw={0,0,-1,1};

        for(int i=0;i<4;i++){
            int h_check=h+dh[i];
            int w_check=w+dw[i];

            if(h_check>=0 && h_check< board.length
                    && w_check>=0 && w_check<board.length){
                if(board[h_check][w_check].equals(color)) cnt++;
            }
        }
    return cnt;
    }
}